### PR TITLE
employee dao

### PR DESCRIPTION
**Pull request by boazco**
ive add 2 functions that I know I will need (getDepartmentById, getRoleById)
and createEmployee which i need for testing

### CODE REVIEW

File: src/main/kotlin/com/hibob/feedback/dao/EmployeesDao.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import org.springframework.stereotype.Repository
+import java.util.*
+
+@Repository
+class EmployeesDao(private val sql: DSLContext) {
+    private val employeesTable = EmployeesTable.instance
+
+    fun createEmployee(firstName: String, lastName: String, role: String, companyId: UUID, department: String): UUID{
+        return sql.insertInto(employeesTable)
+            .set(employeesTable.employeeId, UUID.randomUUID())
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/feedback/dao/EmployeesDao.kt]: autogenerate

File: src/main/kotlin/com/hibob/feedback/dao/EmployeesDao.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    fun getDepartmentById(employeeId: UUID, companyId: UUID): String {
+        return sql.select(employeesTable.department)
+            .from(employeesTable)
+            .where(employeesTable.employeeId.eq(employeeId))
+            .and(employeesTable.companyId.eq(companyId))
+            .fetchOne()?.let { it[employeesTable.department] }
+            ?: throw BadRequestException("No employee with that id")
+    }
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/feedback/dao/EmployeesDao.kt]: i don't think that you'll need this, let's leave this off until later

File: src/main/kotlin/com/hibob/feedback/dao/Tables.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    val employeeId = createUUIDField("employee_id")
+    val creationDate = createDateField("creation_date")
+    val companyId = createUUIDField("company_id")
+    val status = createVarcharField("status")
+    val feedbackMessage = createVarcharField("feedback_message")
+
+    companion object {
+        val instance = FeedbackTable("feedback")
+    }
+}
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/feedback/dao/Tables.kt:18]: in general i put these classes in the dao, but your call

**Comment by elijulian** [comment]
note that the feedback dao appears here, that's probably because you branched off of that branch. so two things:
1. i don't think that there's any dependency between the two daos, so it would have been best to branch off master
2. if you did branch off the feedback dao branch, set that as the base of this pr so that those changes don't appear here

File: src/main/kotlin/com/hibob/feedback/dao/EmployeesDao.kt (hunk 1 above)
Comment (boazco) [reply src/main/kotlin/com/hibob/feedback/dao/EmployeesDao.kt]: what do you mean by that?
thats the table code
id UUID primary key default gen_random_uuid()
so you dont want me to genrate it in the dao?

**Review by elijulian — APPROVED** [review]

**Pull request merged**